combat-trainer.lic

```diff
@@ -275,36 +282,68 @@ class LootProcess
 
   def dispose_body(game_state)
     return unless @loot_bodies
-    return if DRRoom.dead_npcs.empty?
+    if DRRoom.dead_npcs.empty?
+      @last_ritual = nil
+      return
+    end
     return if Time.now - @loot_timer < @loot_delay
 
     game_state.mob_died = true
-    arrange_mob(DRRoom.dead_npcs.first, game_state)
-    if @necro
+    healing = game_state.casting_consume || game_state.prepare_consume
+    if @necro && @last_ritual.nil?
+      if @heal
+        wounds = check_health
+        echo "Severity to Wounds: #{wounds}" if $debug_mode_ct
+        if !wounds.empty?
+          game_state.prepare_consume = true
+          healing = true
+          game_state.healing_target = wounds[wounds.keys.max].first
+          check_thanatology(DRRoom.dead_npcs.first, 'preserve', game_state)
+          check_thanatology(DRRoom.dead_npcs.first, 'consume', game_state)
+        end
+      end
       if @ritual_type == 'harvest'
         check_thanatology(DRRoom.dead_npcs.first, 'preserve', game_state)
       end
-      check_thanatology(DRRoom.dead_npcs.first, @ritual_type, game_state)
-      check_skinning(DRRoom.dead_npcs.first, game_state) if ['preserve'].include?(@ritual_type)
+      check_thanatology(DRRoom.dead_npcs.first, @ritual_type, game_state) if !healing
+    if ['preserve'].include?(@ritual_type) && !healing
+      arrange_mob(DRRoom.dead_npcs.first, game_state)
+      check_skinning(DRRoom.dead_npcs.first, game_state)
+    end
     elsif @skin
+      arrange_mob(DRRoom.dead_npcs.first, game_state)
       check_skinning(DRRoom.dead_npcs.first, game_state)
     end
-    while 'and get ready to search it' == bput('loot', 'You search', 'I could not find what you were referring to', 'and get ready to search it')
-      pause
-      waitrt?
+    unless healing
+      while 'and get ready to search it' == bput('loot', 'You search', 'I could not find what you were referring to', 'and get ready to search it')
+        @last_ritual = nil
+        pause
+        waitrt?
+      end
     end
     @loot_timer = Time.now
   end
 
-  # partially completed, only works for a single ritual.
+  def consume(mob_noun, game_state, wounds)
+    check_thanatology(DRRoom.dead_npcs.first, 'consume', game_state)
+    if game_state.casting
+      bput('release spell', '')
```

Comment:
Thats strange, I had removed that whole method (i dont believe its used anymore)